default.nix

```diff
@@ -339,16 +350,27 @@ in rec {
         mv $NIX_BUILD_TOP/temp "$PWD/deps/${pname}"
         cd $PWD
 
-        if [ -d ${deps}/deps/${pname}/node_modules ]; then
-          cp -r ${deps}/deps/${pname}/node_modules "deps/${pname}/node_modules"
+        ${if useWorkspace then ''
+          if [ -d ${deps}/deps/${pname}/node_modules ]; then
+            cp -r ${deps}/deps/${pname}/node_modules "deps/${pname}/node_modules"
+            chmod -R +w "deps/${pname}/node_modules"
+          fi
+
+          cp -r $node_modules node_modules
+          chmod -R +w node_modules
+        '' else ''
+          cp -r $node_modules deps/${pname}/node_modules
           chmod -R +w "deps/${pname}/node_modules"
-        fi
-        cp -r $node_modules node_modules
-        chmod -R +w node_modules
+
+          mkdir node_modules
+        ''}
 
         ${linkDirFunction}
 
         linkDirToDirLinks "$(dirname node_modules/${pname})"
+        ${lib.optionalString (! useWorkspace) ''
+          ln -sT "../deps/${pname}" "node_modules/${pname}"
```

Comment:
This was the line you eliminated in cc78373f359d17af2bbdf625338932c8ec851ae5

Reply:
So they weren't using workspaces at some point?

Reply:
Or workspaces didn't make symlinks at top level to constituent packages?  I'm not really sure.